### ISSUE TITLE
error with run_pipeline_test.sh 

### ISSUE THREAD

**Issue by Ramanandan** [marbl/metAMOS#194]
Dear Treangen/metAMOS support team,

After running the commnad "$ python INSTALL.py iMetAMOS".  I followed the instructions in "2.2.4
Running the test suit" 
$ cd ./Test
$ ./run_pipeline_test.sh

I encountered the following error. In my local machine, I have 24GB RAM. 

prabhakaranra@prabhakaranra-P8X58-AS:~/metAMOS-1.5rc3/Test$ ./run_pipeline_test.sh 
Error: cannot find BLAST DB directory, expected it in /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB/. Disabling blastdb dependent programs
Project directory already exists, please specify another
Alternatively, use runPipeline to run an existing project
Error: cannot find BLAST DB directory, expected it in /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB/. Disabling blastdb dependent programs
Error: cannot find BLAST DB directory, expected it in /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB/. Disabling blastdb dependent programs
**no blast DB directory available, disabling steps requiring BLAST DB
[Steps to be skipped]:  set(['FunctionalAnnotation', 'FindRepeats'])
Starting Task = runpipeline.RUNPIPELINE
Starting metAMOS pipeline
Error: cannot find BLAST DB directory, expected it in /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB/. Disabling blastdb dependent programs
Warning: Celera Assembler is not found, some functionality will not be available
Warning: BLASR is not found, some functionality will not be available
Warning: Newbler is not found, some functionality will not be available
Warning: MetaGeneMark is not found, some functionality will not be available
Warning: SignalP+ is not found, some functionality will not be available
Warning: metaphylerClassify is not found, some functionality will not be available
Warning: PHmmer is not found, some functionality will not be available
Warning: PhyloSift was not found, will not be available

Warning: FRCbam is not found, some functionality will not be available
Warning: MPI is not available, some functionality may not be available
[Available RAM: 19 GB]
[	There is *19 GB of RAM currently available on this machine, suggested minimum of 32 GB
[	*Enabling low MEM mode, might slow down some steps in pipeline
[	*If more RAM is available than what is listed above, please close down other programs and restart runPipeline

________________________________________
Tasks which will be run:

Task = assemble.SplitAssemblers
Task = assemble.Assemble
Task = assemble.CheckAsmResults
Task = assemble.SplitMappers
Task = mapreads.MapReads
Task = mapreads.CheckMapResults
Task = mapreads.SplitForORFs
Task = findorfs.FindORFS
Task = validate.Validate
Task = findreps.FindRepeats
Task = annotate.Annotate
Task = fannotate.FunctionalAnnotation
Task = scaffold.Scaffold
Task = findscforfs.FindScaffoldORFS
Task = abundance.Abundance
Task = propagate.Propagate
Task = classify.Classify
Task = postprocess.Postprocess
________________________________________
Warning: Graphviz is not found, some functionality will not be available
metAMOS configuration summary:
metAMOS Version:	v1.5rc2 "Praline Brownie"  workflows: core,imetamos
Time and Date:		2015-04-10
Working directory:	/home/prabhakaranra/metAMOS-1.5rc3/Test/test1
Prefix:			proba
K-Mer:			55
Threads:		15
Taxonomic level:	class
Verbose:		False
Steps to skip:		MultiAlign, FunctionalAnnotation, FindRepeats
Steps to force:		Abundance, FindORFS, Annotate, Propagate, MapReads, Assemble, FindScaffoldORFS, Classify

[citation]
MetAMOS
	Treangen, TJ ⇔  Koren, S, Sommer, DD, Liu, B, Astrovskaya, I, Ondov, B, Darling AE, Phillippy AM, Pop, M. MetAMOS: a modular and open source metagenomic assembly and analysis pipeline. Genome biology, 14(1), R2, 2013.

Step-specific configuration:
[abundance]
MetaPhyler
	
	Liu B, Gibbons T, Ghodsi M, Treangen T, Pop M. Accurate and fast estimation of taxonomic profiles from metagenomic shotgun sequences. BMC Genomics. 2011;12 Suppl 2:S4. Epub 2011 Jul 27.

[multialign]
M-GCAT
	/home/prabhakaranra/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64
	Treangen TJ, Messeguer X. M-GCAT: interactively and efficiently constructing large-scale multiple genome comparison frameworks in closely related species. BMC Bioinformatics, 2006.

[fannotate]
BLAST
	/home/prabhakaranra/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64
	Altschul SF, Gish W, Miller W, Myers EW, Lipman DJ. Basic local alignment search tool. J Mol Biol. 1990 Oct 5;215(3):403-10.

[scaffold]
Bambus 2
	/home/prabhakaranra/metAMOS-1.5rc3/AMOS/Linux-x86_64/bin
	Koren S, Treangen TJ, Pop M. Bambus 2: scaffolding metagenomes. Bioinformatics 27(21): 2964-2971 2011.

[findorfs]
FragGeneScan
	/home/prabhakaranra/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64
	Rho M, Tang H, Ye Y: FragGeneScan: predicting genes in short and error-prone reads. Nucleic Acids Research 2010, 38:e191-e191.

[annotate]
Kraken
	/home/prabhakaranra/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/kraken/bin
	Wood DE, Salzberg SL: Kraken: ultrafast metagenomic sequence classification using exact alignments. Genome Biology 2014, 15:R46.

[assemble]
SOAPdenovo
	/home/prabhakaranra/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64
	Li Y, Hu Y, Bolund L, Wang J: State of the art de novo assembly of human genomes from massively parallel sequencing data.Human genomics 2010, 4:271-277.

[mapreads]
Bowtie
	/home/prabhakaranra/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64
	Langmead B, Trapnell C, Pop M, Salzberg SL. Ultrafast and memory-efficient alignment of short DNA sequences to the human genome. Genome Biol. 2009;10(3):R25. Epub 2009 Mar 4.

[preprocess]
metAMOS built-in filtering
	N/A

[validate]
LAP
	/home/prabhakaranra/metAMOS-1.5rc3/LAP
	Ghodsi M, Hill CM, Astrovskaya I, Lin H, Sommer DD, Koren S, Pop M. De novo likelihood-based measures for comparing genome assemblies. BMC research notes 6:334, 2013.

[other]
Krona
	/home/prabhakaranra/metAMOS-1.5rc3/KronaTools/bin
	Ondov BD, Bergman NH, Phillippy AM. Interactive metagenomic visualization in a Web browser. BMC Bioinformatics. 2011 Sep 30;12:385.

sh: 1: Syntax error: Bad fd number
Starting Task = assemble.ASSEMBLE
    Job  = [preprocess.success -> *.run] completed
Completed Task = assemble.SplitAssemblers
    Job  = [soapdenovo.55.run -> soapdenovo.55.asm.contig] completed
Completed Task = assemble.Assemble
    Job  = [[soapdenovo.55.asm.contig] -> [assemble.ok]] completed
Completed Task = assemble.CheckAsmResults
Uptodate Task = assemble.SplitMappers
 
Starting Task = mapreads.MAPREADS
    Job  = [soapdenovo.55.asm.contig -> soapdenovo.55.contig.cvg] completed
Completed Task = mapreads.MapReads
    Job  = [[soapdenovo.55.contig.cvg] -> [mapreads.ok]] completed
Completed Task = mapreads.CheckMapResults
Uptodate Task = mapreads.SplitForORFs
Starting Task = findorfs.FINDORFS
    Job  = [soapdenovo.55.contig.cvg -> soapdenovo.55.faa] completed
Completed Task = findorfs.FindORFS
Starting Task = validate.VALIDATE
    Job  = [[soapdenovo.55.faa] -> [validate.ok]] completed
Completed Task = validate.Validate
Starting Task = findrepeats.FINDREPEATS
    Job  = [proba.fna -> proba.repeats] completed
Completed Task = findreps.FindRepeats
Starting Task = annotate.ANNOTATE
[*****************************************************************
*************************ERROR***********************************
During annotate, the following command failed with return code 2:
>> /home/prabhakaranra/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/kraken/bin/kraken --threads 15 -db /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB//kraken /home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Annotate/in/proba.asm.contig --preload   | kraken-filter --db /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB//kraken --threshold 0.05 > /home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Annotate/out/proba.hits

*************************DETAILS***********************************
Last 10 commands run before the error (/home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Logs/COMMANDS.log)
|2015-04-10 17:26:43| touch /home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Logs/findrepeats.skip
|2015-04-10 17:26:44|# [ANNOTATE]
|2015-04-10 17:26:44| touch /home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Annotate/out/proba.annots
|2015-04-10 17:26:44| unlink /home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Annotate/in/proba.asm.contig
|2015-04-10 17:26:44| ln /home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Assemble/out/proba.asm.contig /home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Annotate/in/proba.asm.contig
|2015-04-10 17:26:44| unlink /home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Annotate/out/proba.hits
|2015-04-10 17:26:44| rm -f /home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Annotate/out/*.hits
|2015-04-10 17:26:44| rm -f /home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Annotate/out/*.epsilon-nb_results.txt
|2015-04-10 17:26:44| rm -f /home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Annotate/out/*.phymm.out
|2015-04-10 17:26:44| /home/prabhakaranra/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/kraken/bin/kraken --threads 15 -db /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB//kraken /home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Annotate/in/proba.asm.contig --preload   | kraken-filter --db /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB//kraken --threshold 0.05 > /home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Annotate/out/proba.hits

Last 10 lines of output (/home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Logs/ANNOTATE.log)
unlink: cannot unlink ‘/home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Annotate/out/proba.hits’: No such file or directory
kraken-filter: can't open nodes file: No such file or directory
kraken: /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB//kraken/database.kdb does not exist!

Please veryify input data and restart MetAMOS. If the problem persists please contact the MetAMOS development team.
*************************ERROR***********************************
*****************************************************************
rm: cannot remove ‘/home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Logs/annotate.ok’: No such file or directory
Oops, MetAMOS finished with errors! see text in red above for details.
prabhakaranra@prabhakaranra-P8X58-AS:~/metAMOS-1.5rc3/Test$

**Comment by skoren** [comment]
The error indicates you are missing the Kraken database. Does the listed file (/home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB//kraken/) exist? I've verified the download link in INSTALL.py is correct and working.I'm guessing for the same reason that your pull from git was failing, your system rejects some outgoing connections (Kraken's DB is downloaded over http). You can try manually downloading the Kraken mini DB from http://ccb.jhu.edu/software/kraken/dl/minikraken.tgz and extracting it to /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB//kraken/:

```
tar xvzf minikraken.tgz
mv minikraken_* /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB/kraken
```

**Comment by Ramanandan** [comment]
Dear Serge Koren,

Thanks for pointing out the issues. When I checked this path (/home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB//kraken/), I had only the database.idx file. 

As you sugested, I downloaded the file and move it to Kraken folder. Then I executed the below mentioned test suite commands. I received few error and warning messages. I pasted the entire comments from the pipeline for your reference. Seems like the program executed properly. But I couldn't see the anything in html output. I hope that is because of the warning listed below like Celera Assembler, BLASR, Newbler, MetaGeneMark, SignalP+, metaphylerClassify, PHmmer, PhyloSift,MPI  and,FRCbam  are not found. Please, correct me if I am wrong?

prabhakaranra@prabhakaranra-P8X58-AS:~/metAMOS-1.5rc3$ tar xvzf minikraken.tgz
minikraken_20141208/
minikraken_20141208/database.idx
minikraken_20141208/database.kdb
minikraken_20141208/taxonomy/
minikraken_20141208/taxonomy/nodes.dmp
minikraken_20141208/taxonomy/names.dmp
minikraken_20141208/MiniKraken.README
prabhakaranra@prabhakaranra-P8X58-AS:~/metAMOS-1.5rc3$ mv minikraken_*/ /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB/kraken
prabhakaranra@prabhakaranra-P8X58-AS:~/metAMOS-1.5rc3$ cd ./Test/
prabhakaranra@prabhakaranra-P8X58-AS:~/metAMOS-1.5rc3/Test$ ./run_pipeline_test.sh 
Error: cannot find BLAST DB directory, expected it in /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB/. Disabling blastdb dependent programs
Project directory already exists, please specify another
Alternatively, use runPipeline to run an existing project
Error: cannot find BLAST DB directory, expected it in /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB/. Disabling blastdb dependent programs
Error: cannot find BLAST DB directory, expected it in /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB/. Disabling blastdb dependent programs
**no blast DB directory available, disabling steps requiring BLAST DB
[Steps to be skipped]:  set(['FunctionalAnnotation', 'FindRepeats'])
Starting Task = runpipeline.RUNPIPELINE
Starting metAMOS pipeline
Error: cannot find BLAST DB directory, expected it in /home/prabhakaranra/metAMOS-1.5rc3/Utilities/DB/. Disabling blastdb dependent programs
Warning: Celera Assembler is not found, some functionality will not be available
Warning: BLASR is not found, some functionality will not be available
Warning: Newbler is not found, some functionality will not be available
Warning: MetaGeneMark is not found, some functionality will not be available
Warning: SignalP+ is not found, some functionality will not be available
Warning: metaphylerClassify is not found, some functionality will not be available
Warning: PHmmer is not found, some functionality will not be available
Warning: PhyloSift was not found, will not be available

Warning: FRCbam is not found, some functionality will not be available
Warning: MPI is not available, some functionality may not be available
[Available RAM: 17 GB]
#[	There is *17 GB of RAM currently available on this machine, suggested minimum of 32 GB
#[	*Enabling low MEM mode, might slow down some steps in pipeline
#[	*If more RAM is available than what is listed above, please close down other programs and restart runPipeline

________________________________________
Tasks which will be run:

Task = assemble.SplitAssemblers
Task = assemble.Assemble
Task = assemble.CheckAsmResults
Task = assemble.SplitMappers
Task = mapreads.MapReads
Task = mapreads.CheckMapResults
Task = mapreads.SplitForORFs
Task = findorfs.FindORFS
Task = validate.Validate
Task = findreps.FindRepeats
Task = annotate.Annotate
Task = fannotate.FunctionalAnnotation
Task = scaffold.Scaffold
Task = findscforfs.FindScaffoldORFS
Task = abundance.Abundance
Task = propagate.Propagate
Task = classify.Classify
Task = postprocess.Postprocess
________________________________________
Warning: Graphviz is not found, some functionality will not be available
metAMOS configuration summary:
metAMOS Version:	v1.5rc2 "Praline Brownie"  workflows: core,imetamos
Time and Date:		2015-04-13
Working directory:	/home/prabhakaranra/metAMOS-1.5rc3/Test/test1
Prefix:			proba
K-Mer:			55
Threads:		15
Taxonomic level:	class
Verbose:		False
Steps to skip:		MultiAlign, FunctionalAnnotation, FindRepeats
Steps to force:		Abundance, FindORFS, Annotate, Propagate, MapReads, Assemble, FindScaffoldORFS, Classify

[citation]
MetAMOS
	Treangen, TJ ⇔  Koren, S, Sommer, DD, Liu, B, Astrovskaya, I, Ondov, B, Darling AE, Phillippy AM, Pop, M. MetAMOS: a modular and open source metagenomic assembly and analysis pipeline. Genome biology, 14(1), R2, 2013.


Step-specific configuration:
[abundance]
MetaPhyler
	
	Liu B, Gibbons T, Ghodsi M, Treangen T, Pop M. Accurate and fast estimation of taxonomic profiles from metagenomic shotgun sequences. BMC Genomics. 2011;12 Suppl 2:S4. Epub 2011 Jul 27.

[multialign]
M-GCAT
	/home/prabhakaranra/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64
	Treangen TJ, Messeguer X. M-GCAT: interactively and efficiently constructing large-scale multiple genome comparison frameworks in closely related species. BMC Bioinformatics, 2006.

[fannotate]
BLAST
	/home/prabhakaranra/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64
	Altschul SF, Gish W, Miller W, Myers EW, Lipman DJ. Basic local alignment search tool. J Mol Biol. 1990 Oct 5;215(3):403-10.

[scaffold]
Bambus 2
	/home/prabhakaranra/metAMOS-1.5rc3/AMOS/Linux-x86_64/bin
	Koren S, Treangen TJ, Pop M. Bambus 2: scaffolding metagenomes. Bioinformatics 27(21): 2964-2971 2011.

[findorfs]
FragGeneScan
	/home/prabhakaranra/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64
	Rho M, Tang H, Ye Y: FragGeneScan: predicting genes in short and error-prone reads. Nucleic Acids Research 2010, 38:e191-e191.

[annotate]
Kraken
	/home/prabhakaranra/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/kraken/bin
	Wood DE, Salzberg SL: Kraken: ultrafast metagenomic sequence classification using exact alignments. Genome Biology 2014, 15:R46.

[assemble]
SOAPdenovo
	/home/prabhakaranra/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64
	Li Y, Hu Y, Bolund L, Wang J: State of the art de novo assembly of human genomes from massively parallel sequencing data.Human genomics 2010, 4:271-277.

[mapreads]
Bowtie
	/home/prabhakaranra/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64
	Langmead B, Trapnell C, Pop M, Salzberg SL. Ultrafast and memory-efficient alignment of short DNA sequences to the human genome. Genome Biol. 2009;10(3):R25. Epub 2009 Mar 4.

[preprocess]
metAMOS built-in filtering
	N/A

[validate]
LAP
	/home/prabhakaranra/metAMOS-1.5rc3/LAP
	Ghodsi M, Hill CM, Astrovskaya I, Lin H, Sommer DD, Koren S, Pop M. De novo likelihood-based measures for comparing genome assemblies. BMC research notes 6:334, 2013.

[other]
Krona
	/home/prabhakaranra/metAMOS-1.5rc3/KronaTools/bin
	Ondov BD, Bergman NH, Phillippy AM. Interactive metagenomic visualization in a Web browser. BMC Bioinformatics. 2011 Sep 30;12:385.


sh: 1: Syntax error: Bad fd number
Starting Task = assemble.ASSEMBLE
    Job  = [preprocess.success -> *.run] completed
Completed Task = assemble.SplitAssemblers
    Job  = [soapdenovo.55.run -> soapdenovo.55.asm.contig] completed
Completed Task = assemble.Assemble
    Job  = [[soapdenovo.55.asm.contig] -> [assemble.ok]] completed
Completed Task = assemble.CheckAsmResults
Uptodate Task = assemble.SplitMappers
 
Starting Task = mapreads.MAPREADS
    Job  = [soapdenovo.55.asm.contig -> soapdenovo.55.contig.cvg] completed
Completed Task = mapreads.MapReads
    Job  = [[soapdenovo.55.contig.cvg] -> [mapreads.ok]] completed
Completed Task = mapreads.CheckMapResults
Uptodate Task = mapreads.SplitForORFs
Starting Task = findorfs.FINDORFS
    Job  = [soapdenovo.55.contig.cvg -> soapdenovo.55.faa] completed
Completed Task = findorfs.FindORFS
Starting Task = validate.VALIDATE
    Job  = [[soapdenovo.55.faa] -> [validate.ok]] completed
Completed Task = validate.Validate
Starting Task = findrepeats.FINDREPEATS
    Job  = [proba.fna -> proba.repeats] completed
Completed Task = findreps.FindRepeats
Starting Task = annotate.ANNOTATE
    Job  = [proba.faa -> proba.hits] completed
Completed Task = annotate.Annotate
Starting Task = functionalannotation.FUNCTIONALANNOTATION
    Job  = [proba.faa -> [blast.out, krona.ec.input]] completed
Completed Task = fannotate.FunctionalAnnotation
Starting Task = scaffold.SCAFFOLD
    Job  = [[proba.asm.contig] -> scaffold.ok] completed
Completed Task = scaffold.Scaffold
Starting Task = findscaffoldorfs.FINDSCAFFOLDORFS
    Job  = [proba.linearize.scaffolds.final -> proba.fna] completed
Completed Task = findscforfs.FindScaffoldORFS
Starting Task = abundance.ABUNDANCE
    Job  = [proba.asm.contig -> proba.taxprof.pct.txt] completed
Completed Task = abundance.Abundance
Starting Task = propagate.PROPAGATE
    Job  = [proba.annots -> propagate.ok] completed
Completed Task = propagate.Propagate
Starting Task = classify.CLASSIFY
    Job  = [proba.clusters -> classify.ok] completed
Completed Task = classify.Classify
Starting Task = postprocess.POSTPROCESS
    Job  = [proba.asm.contig -> proba.scf.fa] completed
Completed Task = postprocess.Postprocess
done! pipeline took 1.37 minutes
Please see results in /home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Postprocess/out
metAMOS summary report is file:///home/prabhakaranra/metAMOS-1.5rc3/Test/test1/Postprocess/out/html/summary.html
prabhakaranra@prabhakaranra-P8X58-AS:~/metAMOS-1.5rc3/Test$ 
(process:27476): GLib-CRITICAL **: g_slice_set_config: assertion 'sys_page_size == 0' failed
^C
prabhakaranra@prabhakaranra-P8X58-AS:~/metAMOS-1.5rc3/Test$

**Comment by skoren** [comment]
All the warnings are fine. Those are packages that have licensing restrictions that prevent us from re-distributing them or installing them directly by metAMOS (i.e. Newbler comes from Roche and is not open-source). A couple of the tools (PHMMER, PhyloSift) are optional and only installed if you run python INSTALL.py optional.

You opened the generated HTML file and nothing was displayed? What browser are you using (version/name) and is the HTML file non-0 sized?

**Comment by Ramanandan** [comment]
Dear Dr. Sergey Koren,

1) In my local machine, I have both firefox (firefox 36.0.1+build2-0ubuntu0.14.04.1) and google chrome (Version 42.0.2311.90 (64-bit)).
![local machine test run output](https://cloud.githubusercontent.com/assets/11878969/7144599/a017a868-e2b4-11e4-87ac-2df8dda54b26.png)



2) In my workstation, only firefox (Mozilla Firefox 37.0.1).

![Uploading Workstation test run output.png . . .]()

**Comment by Ramanandan** [comment]
Dear Dr. Sergey Koren,

1) In my local machine, I have both firefox (firefox 36.0.1+build2-0ubuntu0.14.04.1) and google chrome (Version 42.0.2311.90 (64-bit)).
![local machine test run output](https://cloud.githubusercontent.com/assets/11878969/7144599/a017a868-e2b4-11e4-87ac-2df8dda54b26.png)



2) In my workstation, only firefox (Mozilla Firefox 37.0.1).

![workstation test run output](https://cloud.githubusercontent.com/assets/11878969/7144618/bae26fe8-e2b4-11e4-8c53-4d764de2ca43.png)

**Comment by skoren** [comment]
Both of those results look correctly populated. You can use the tabs at left to navigate to view the results of each step. The preprocess step (which is selected) just summarizes your input data.

**Comment by Ramanandan** [comment]
Dear Sergey Koran,

Please find some of the tabs I selected and pasted the screenshot for your reference.

![assemble tab is selected](https://cloud.githubusercontent.com/assets/11878969/7146215/0441abda-e2c0-11e4-8f3b-9754029e8ea1.png)

**Comment by Ramanandan** [comment]
How do I unopen this issue again. I could see only comment button from my side.

**Comment by skoren** [comment]
Does the display work in Firefox. There was an issue with chrome due to the way it restricts local javascript (it doesn't allow the javascript to read the local input for the chart). This issue was fully fixed in rc3 but since you're in rc2 you may still be seeing it.

**Comment by Ramanandan** [comment]
The below screenshot from firefox webpage. The webpages are still blank. 

![screenshot from firefox](https://cloud.githubusercontent.com/assets/11878969/7147160/d3e551a6-e2c6-11e4-83d7-e7647811382b.png)

**Comment by Ramanandan** [comment]
Dear Dr. Sergey Koren,

I copied the test1 folder to other systems and tried opening the results with different web browser. No luck, still the graphs are not displayed. 

Each and every time when I execute run_pipeline_test.sh, I see this error displayed "sh: 1: Syntax error: Bad fd number". Does it interrupt with my html files?

Do you have the expected output files and html files for command run_pipeline_test.sh? 

If it is available, I can check what are all the files are same between my output files and reference output files.

**Comment by skoren** [comment]
The screenshots you posted above do look correct. The Annotate/Abundance will be blank for this dataset because this genome doesn't have the marker genes metaphyler is using and Kraken cannot confidently classify it with the mini database (it can with the full one). This is why you have all three contigs listed as unclassified. If you run one of the larger test cases (like test_sra.sh) there should be some output in the Annotate plot.

The Assemble/MapReads should not be blank so you must be missing the data files that feed the HTML pages. These are in Postprocess/out/html:

```
bash-3.2$ cat test1/Postprocess/out/html/mapreads.js 
var contiglens = "cov	frequency\n" + 
"45.27	1.000000\n" + 
"45.51	0.000000\n" + 
"45.76	0.000000\n" + 
"46.00	0.000000\n" + 
"46.25	0.000000\n" + 
"46.49	0.000000\n" + 
"46.73	0.000000\n" + 
"46.98	0.000000\n" + 
"47.22	0.000000\n" + 
"47.47	0.000000\n" + 
"47.71	0.000000\n" + 
"47.95	0.000000\n" + 
"48.20	0.000000\n" + 
"48.44	0.000000\n" + 
"48.69	0.000000\n" + 
"48.93	0.000000\n" + 
"49.17	0.000000\n" + 
"49.42	0.000000\n" + 
"49.66	0.000000\n" + 
"49.91	2.000000\n" + 
"";bash-3.2$ cat test1/Postprocess/out/html/assemble.js 
var contiglens = "total	contig\n" + 
"92399	92399\n" + 
"158102	65703\n" + 
"159795	1693\n" + 
```

Can you check if you have those files in your run and what their contents are?  As for the scaffold plot, during installation did you have an error about matplot lib? If that library is missing the scaffold plot cannot be generated. You'd have to install matplotlib 1.1.0 yourself for python to enable it if that's the case.

**Comment by Ramanandan** [comment]
Dear Dr. Sergey Koren,
 
I don't have any of these javascript files in my html folder. 
So what I did,
1)  rerun "$ python INSTALL.py iMetAMOS". I noticed the following error: "
In file included from src/ft2font.cpp:3:0:src/ft2font.h:16:22: fatal error: ft2build.h: No such file or directory #include <ft2build.h>"
2) I googled, and found this "http://stackoverflow.com/questions/762292/matplotlib-build-problem-error-c1083-cannot-open-include-file-ft2build-h"
3) I executed following command - sudo apt-get install python-dev libfreetype6-dev
4) re-run  "$ python INSTALL.py iMetAMOS". The issue got resolved.
5) executed ./run_pipeline_test.sh
6) Now I could see assemnle.js, mapreads.js files in my folder
6) finally it worked.

Wow :)!!, now I could see the graph/chart in a) Assemble, b)MapReads, c) FindORFs, d) Scaffold, e) Findscaffoldorfs. As you mentioned, I din't see chart in 1) Annotate, 2) Abundance.
